src/main/scala/cromwell/engine/workflow/ValidateActor.scala

```diff
@@ -10,65 +15,181 @@ import wdl4s._
+        validatedBackends <- executableBackendsForWf(namespaceWithWorkflow, Option(coercedInputs), workflowOptions)
+      } yield ValidationSuccess(namespaceWithWorkflow,
+        Option(coercedInputs),
+        workflowOptions,
+        validatedBackends,
+        Option(WorkflowSourceFiles(wdlSource, workflowInputs.getOrElse("{}"), workflowOptions.getOrElse("{}"))))
+
+      futureValidationOutcome onComplete {
+        case Success(validationOutcome: ValidationSuccess) =>
+          requster ! validationOutcome
+        case Failure(reason) =>
+          requster ! ValidationFailure(reason)
+      }
+  }
+
+  private def buildCoercedInputs(namespaceWithWorkflow: NamespaceWithWorkflow, rawInputs: Option[WdlJson]): Future[WorkflowCoercedInputs] = {
+    val inputs = workflowInputs.get.parseJson.asJsObject.fields
+    Future.fromTry(namespaceWithWorkflow.coerceRawInputs(inputs))
   }
 
+  /**
+    * This method will validate the workflow with all the available backends. The WF fails
+    * validation if this method returns a list of size 0, which meant that no backend was willing to accept
+    * this workflow.
+    * Later on, this method can be changed to a 'per-call-basis' acceptance test by the backends,
+    * i.e. it may return a Map of [Call -> ExecutableBackends]
+    *
+    * @param namespace     workflow with the namespaces already resolved
+    * @param wfInputs      Already coerced inputs
+    * @param wfOptionsJson workflow options
+    * @return List of BackendConfigurationEntry objects, representing which backends can accept this WF
+    */
+  private def executableBackendsForWf(namespace: NamespaceWithWorkflow,
+                                      wfInputs: Option[WorkflowCoercedInputs] = None,
+                                      wfOptionsJson: Option[String] = None): Future[Seq[BackendConfigurationEntry]] = {
+    val listOfPluggedBackends: List[BackendConfigurationEntry] = BackendConfiguration.apply().getAllBackendConfigurations()
+
+    logger.info(s"Plugged in Backends: $listOfPluggedBackends")
+
+    //If the validation class is not specified, it is assumed that backend will accept any WF
+    // The `willExecute` list contains backends that don't have a validation class, and will accept this workflow
+    val (mayExecute, willExecute) = listOfPluggedBackends.partition(_.validationClass.isDefined)
+
+    if (willExecute.nonEmpty) logger.info(s"Backends with no validation requirements: $willExecute")
+
+    val backendEntryValidatorActorPair: Map[BackendConfigurationEntry, ActorRef] = mayExecute map { entry =>
+      entry -> DefaultBackendFactory.getBackendActorFor(entry.validationClass.get, context.system, namespace, wfInputs, wfOptionsJson)
+    } toMap
+
+    // The implicit timeout required for Ask below comes from `CromwellActor`
+    val mapOfEntryToFutureValidationRes: Map[BackendConfigurationEntry, Future[ValidationResult]] = backendEntryValidatorActorPair mapValues { validateActor =>
```

Comment:
looks like these two vals (backendEntryValidatorActorPair,mapOfEntryToFutureValidationRes) can be combined like : f1 andThen f2